GTEx/CAVATICA RNA-Seq Training Documentation

HigginsD3b opened the pull request:
## PR Checklist

- [x] Release label added
- [X] PR category label added
- [X] PR mergeable
- [X] Clean build logs
- [X] Linked related issues
- [X] Colorblindness test https://www.toptal.com/designers/colorfilter/

## PR Description
__Briefly describe the changes this PR will add/address__
This PR will add an additional page to the RNA-Seq on CAVATICA training documents. This PR demonstrates how to identify GTEx Datasets from the AnVIL Portal and import them into the CAVATICA/Kids First environment. This demonstration builds on the existing RNA-Seq training documentation by providing directions for expanding an analysis to include GTEx's data as a source of RNA-Seq controls to compare to Kids First tumor samples. It also demonstrates interoperability between two of CFDE's DCCs.

Summary of Changes:
- Created [rna-seq-9.md](https://github.com/nih-cfde/training-and-engagement/blob/Cavatica-RNA-seq-training-updates/docs/Bioinformatic-Analyses/RNAseq-on-Cavatica/rna_seq_9.md), the new tutorial page.
- Added screenshot assets to the `rna-seq-images` directory in support of the new page.
- Updated the [.pages nav bar](https://github.com/nih-cfde/training-and-engagement/blob/Cavatica-RNA-seq-training-updates/docs/Bioinformatic-Analyses/RNAseq-on-Cavatica/.pages) to include the new page.

**Preview link**
I am not familiar with how to generate this link, though the markdown page can be viewed from the link above.

## Review format
Direct editing is fine for small comments. Please comment with any larger changes requested.

## Timeline
If there are no major concerns with the content of the lesson, we hope to have this merged ASAP to complete a CFDE milestone and fulfill a request from George Papanicolaou. cc: PM @ericwenger-pm who has been supporting this project.

HigginsD3b committed:
Create rna_seq_9.md

HigginsD3b committed:
Update rna_seq_9.md

HigginsD3b committed:
Update rna_seq_9.md

HigginsD3b committed:
Update rna_seq_9.md

HigginsD3b committed:
Update rna_seq_9.md

HigginsD3b committed:
Update rna_seq_9.md

HigginsD3b committed:
Update rna_seq_9.md

HigginsD3b committed:
Add files via upload

HigginsD3b committed:
Delete rna-seq-9-01.png

HigginsD3b committed:
Delete rna-seq-9-02.png

HigginsD3b committed:
Delete rna-seq-9-03.png

HigginsD3b committed:
Delete rna-seq-9-04.png

HigginsD3b committed:
Delete rna-seq-9-05.png

HigginsD3b committed:
Delete rna-seq-9-06.png

HigginsD3b committed:
Delete rna-seq-9-07.png

HigginsD3b committed:
Delete rna-seq-9-08.png

HigginsD3b committed:
Delete rna-seq-9-09.png

HigginsD3b committed:
Delete rna-seq-9-10.png

HigginsD3b committed:
Delete rna-seq-9-11.png

HigginsD3b committed:
Delete rna-seq-9-12.png

HigginsD3b committed:
Delete rna-seq-9-13.png

HigginsD3b committed:
Delete rna-seq-9-14.png

HigginsD3b committed:
Delete rna-seq-9-15.png

HigginsD3b committed:
Delete rna-seq-9-16.png

HigginsD3b committed:
Delete rna-seq-9-17.png

HigginsD3b committed:
Delete rna-seq-9-18.png

HigginsD3b committed:
Delete rna-seq-9-19.png

HigginsD3b committed:
Delete rna-seq-9-20.png

HigginsD3b committed:
Delete rna-seq-9-21.png

HigginsD3b committed:
Delete rna-seq-9-22.png

HigginsD3b committed:
Delete rna-seq-9-23.png

HigginsD3b committed:
Delete rna-seq-9-24.png

HigginsD3b committed:
Add files via upload

HigginsD3b committed:
Update rna_seq_9.md

HigginsD3b committed:
Update rna_seq_9.md

ericwenger-pm committed:
minor updates to rna_seq_9.md

two minor updates
Section 2
Change from: 
- Once the PFB is created, download it to your local machine. This document [...]

to:
- Once the PFB is created, download it to your local machine. This downloaded PFB file [...]

Section 3
Change from:
The commands for importing the GTEx files from AnVIL into CAVATICA are written into the `interop.py` script.

to:
The commands for importing the GTEx files from AnVIL into CAVATICA are written into the `interop.py` script included in the public project you copied.

HigginsD3b committed:
Update .pages

Adding the new page to the nav bar.

raynamharris commented:
Thanks @HigginsD3b! I'll work through the new tutorial today. 

In the meantime, the preview link can be tricky to find. Do you see where it says "All checks have passed. 4 successful checks. Show all checks."? Click "show all checks" so you can see the details. Right click the "Details" hyperlink next to "docs/readthedocs.com...." and copy the hyperlink (or click Details and it will open in your browser). This is your preview link. Note, it has your PR number. 

![Screen Shot 2022-01-28 at 8 45 06 AM](https://user-images.githubusercontent.com/7544197/151588270-4bab25f4-4295-41a5-87e7-f05305c01845.png)


https://cfde-training-and-engagement--503.com.readthedocs.build/en/503/

At first glance, the images aren't rendering, and I'm not sure why. I would try one of two things. Either replace the `.` with the  full path to here or replace the html formatting with markdown formatting. Both examples are below. 

```
<img src="/docs/Bioinformatic-Analyses/RNAseq-on-Cavatica/rna-seq-images/rna-seq-9-01.png" width="400"/>

![](./rna-seq-images/rna-seq-9-01.png)
 ```

raynamharris commented on docs/Bioinformatic-Analyses/RNAseq-on-Cavatica/rna_seq_9.md:
```diff
@@ -0,0 +1,86 @@
+## Step 1: Identify GTEx Files on AnVIL and Export a PFB
+- Navigate to [the AnVIL Portal](https://gen3.theanvil.io/login) and login using NIH or Google credentials.
+    - <img src="./rna-seq-images/rna-seq-9-01.png" width="400"/>
+- Using the [AnVIL Data Explorer](https://gen3.theanvil.io/explorer), identify the GTEx files for your analysis. For more information on using AnVIL Gen3, see [their support documentation here](https://anvilproject.org/learn/introduction/getting-started-with-gen3).
```
On line 21, I would start the sentences with 'Click the "Explore Data" button to use the AnVIL Data Explorer....

Also, since this is a tutorial, I think you need to tell the reader exactly which GTEx files to identify. This way, they can follow along. At the end of the tutorial, you suggest that they go through it again with their files of interest

ericwenger-pm replied:
added requested change for line 21, Click the "Explore Data" button to use AnVIL Data Explorer

ericwenger-pm replied:
agree that we can get some clarifying language on file choice

raynamharris commented on docs/Bioinformatic-Analyses/RNAseq-on-Cavatica/rna_seq_9.md:
```diff
@@ -0,0 +1,86 @@
+    - <img src="./rna-seq-images/rna-seq-9-01.png" width="400"/>
+- Using the [AnVIL Data Explorer](https://gen3.theanvil.io/explorer), identify the GTEx files for your analysis. For more information on using AnVIL Gen3, see [their support documentation here](https://anvilproject.org/learn/introduction/getting-started-with-gen3).
+    - <img src="./rna-seq-images/rna-seq-9-02.png" width="400"/>
+- Once you have identified the subset of files for your analysis, return to the top of the screen and choose `Export All to PFB`. This will create a manifest of files on the AnVIL Portal. This process may take several minutes, depending upon the number of files for export.
```
On my screen, the button text is `Export to PFB` (without the "All")

![Screen Shot 2022-01-28 at 9 09 58 AM](https://user-images.githubusercontent.com/7544197/151591155-ea23b085-435a-42b0-b8e0-7afc041f256f.png)

ericwenger-pm replied:
line 23 changed `Export All to PFB` to `Export to PFB`

raynamharris commented on docs/Bioinformatic-Analyses/RNAseq-on-Cavatica/rna_seq_9.md:
```diff
@@ -0,0 +1,86 @@
+- Once the PFB is created, download it to your local machine. This downloaded PFB file contains the information CAVATICA needs to pull these files into your CAVATICA project.
+    - <img src="./rna-seq-images/rna-seq-9-04.png" width="400"/>
+- Before navigating away from the AnVIL Portal, you also must generate an API key. This will be used to authenticate your access to the files in the PFB and allow the transfer of GTEx data into CAVATICA. Select your username in the top right of the AnVIL Portal. Select `Create API Key` in the top left of the page that opens. After a key is generated, download this key by choosing `Dowload json`.  ...
+    - <img src="./rna-seq-images/rna-seq-9-06.png" width="400"/>
```
On line 28, the image is a screenshot of the Cavatica home page, but you haven't told the reader how to get there. I think this image would be more appropriate in the next section. If you do include an image here, I think it should be a screenshot of the "Create API key".

ericwenger-pm replied:
corrected image reference on line 28 from a Cavatica login screen on rna-seq-9-06.png to an AnVIL Create API Key

raynamharris commented on docs/Bioinformatic-Analyses/RNAseq-on-Cavatica/rna_seq_9.md:
```diff
@@ -0,0 +1,86 @@
+
+## Step 2: Access the Data Interoperability Jupyter Notebook on CAVATICA
+The commands necessary to import the chosen GTEx files from the AnVIL Portal have been saved in a Jupyter Notebook that is available on the CAVATICA platform. We will copy this notebook to a project we can work in and then begin an interactive session in the Data Cruncher to run the transfer.
+- Navigate to [CAVATICA](https://cavatica.sbgenomics.com/) and login using NIH or CAVATICA credentials.
```
Since this is my first time logging in, I also had to grant consent and authorize something. Would be good to include that as well. 

![Screen Shot 2022-01-28 at 9 25 52 AM](https://user-images.githubusercontent.com/7544197/151593881-c82deebb-9204-4220-bef1-d65b20c7ad06.png)

![Screen Shot 2022-01-28 at 9 26 13 AM](https://user-images.githubusercontent.com/7544197/151593894-43fab082-efc3-489d-bff5-fa716708509a.png)

Unfortunately, this is as far as I can get for now because it says my seven bridges account is not authorized. 

![Screen Shot 2022-01-28 at 9 26 34 AM](https://user-images.githubusercontent.com/7544197/151593906-41971c85-9056-4dd2-83d6-ac4574318685.png)

ericwenger-pm replied:
this likely was temporary account issue, maybe timeout or expired key or something else, did you contact Cavatica support and resolve?

ericwenger-pm replied:
added the NIH consent and Gen3 authorization screenshots with a note that users logging into Cavatica for the first time would be presented with these screens.  Also generally addressed an issue with screenshots for section 2 being off by one, corrected now

raynamharris reviewed (changes requested):
I worked through the first section. As soon as I get my Seven Bridges account reactivated, I'll work through the rest of the tutorial.

raynamharris commented on docs/Bioinformatic-Analyses/RNAseq-on-Cavatica/rna_seq_9.md:
```diff
@@ -0,0 +1,86 @@
+
+## Step 3: Within Data Cruncher, Import the Files in the PFB
+The commands for importing the GTEx files from AnVIL into CAVATICA are written into the `interop.py` script included in the public project you copied.  We will be using the Jupyter Notebook `AnVIL_transfer.ipynb` to input the PFB manifest and authentication for the transfer.
+- Open the `AnVIL_transfer.ipynb` notebook by selecting it from the left column in the Data Cruncher.
```
This step says to open the `AnVIL_transfer.ipynb` notebook, but the photo shows the user opening `gtex...ipynb`, which is a file that is not in my environment.

![Screen Shot 2022-02-03 at 1 58 56 PM](https://user-images.githubusercontent.com/7544197/152436443-af341ea1-e04c-4724-8f33-7f52bd456707.png)

ericwenger-pm replied:
replaced outdated rna-seq-9-14.png with rna-seq-9-14-AnVIL_transfer_ipynb.png

raynamharris commented on docs/Bioinformatic-Analyses/RNAseq-on-Cavatica/rna_seq_9.md:
```diff
@@ -0,0 +1,86 @@
+    - Set `target_project = 'USER/PROJECT'`, where `USER` is your CAVATICA username and `PROJECT` is the name of the CAVATICA project you wish to move the files to. You may also copy these from the URL of the project after navigating there in your browser window.
+    - Set `pfb = 'PATH'`, where `PATH` is the path to the .avro PFB file you uploaded. You can copy the path by right-clicking the file and choosing `Copy Path`. 
+        - <img src="./rna-seq-images/rna-seq-9-20.png" width="400"/>
+- When all three variables have been set, use the `Play` button to run this paragraph. The files will begin to transfer and appear one-by-one below the paragraph.
```
I have set all three variables, but nothing happens when I click the play button. 

```
# Select the platform you are working on
target_platform = 'cavatica' # 'cgc' for CGC, 'bdc' for BioData Catalyst, 'cavatica' for CAVATICA

# Select the project to import the data
target_project = 'rmharris8/copy-of-data-interoperability' # slug from URL: username/project_name

# Path to pfb file, e.g. ./test-data/gtex_analysis_txt.avro
pfb = 'export_2022-01-28T17_09_23.avro'

gtex_records = get_sequencing_records(pfb_file=pfb, client=interop_client)

# Note that this step can take longer for bigger files
for record in gtex_records:
    record.import_to_project(platform=target_platform, project=target_project)
```

raynamharris replied:
update, i had missed adding the 7 bridges token. now, i get an error message when I click play.

```
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
/tmp/ipykernel_204/1619340111.py in <module>
     12 # Note that this step can take longer for bigger files
     13 for record in gtex_records:
---> 14     record.import_to_project(platform=target_platform, project=target_project)

/sbgenomics/workspace/interop.py in import_to_project(self, platform, project)
    364         )
    365         if self.anvil_client.api_key:
--> 366             file_url = self.anvil_client.import_drs(
    367                 self.seq_object.drs_import_link)
    368         else:

/sbgenomics/workspace/interop.py in import_drs(self, drs_link, access_id)
    174         drs_split = without_drs_part.split(':')
    175         drs_id = drs_split[1]
--> 176         drs_object = self.get_file_url(drs_id, access_id)
    177         return drs_object['url']
    178 

/sbgenomics/workspace/interop.py in get_file_url(self, drs_id, access_id)
    162                 drs_id, access_id, r['stef_status_code'], r['message']
    163             )
--> 164             raise Exception('Unable to get file with drs')
    165 
    166         return r

Exception: Unable to get file with drs
```

raynamharris commented:
Hi @HigginsD3b , I made it though the tutorial, but I'm getting an error message on the last step. Any ideas?

ericwenger-pm committed:
Update rna_seq_9.md

added requested change, Click the "Explore Data" button to use AnVIL Data Explorer

ericwenger-pm committed:
Update rna_seq_9.md

line 23 changed `Export All to PFB` to `Export to PFB`

ericwenger-pm committed:
Update rna_seq_9.md

corrected image reference on line 28 from a Cavatica login screen on rna-seq-9-06.png to an AnVIL Create API Key

ericwenger-pm committed:
Update rna_seq_9.md

corrected Step 2 screenshots -- they were offset by one, starting with "Navigate to Cavatica" which needed to be changed from rna-seq-9-07.png to rna-seq-9-06.png up until "clicking to enter Data Cruncher" from rna-seq-9-13.png to rna-seq-9-12.png

ericwenger-pm committed:
first-time-login-consents images

rna-seq-9-06-01-nih-consent.png
rna-seq-9-06-02-gen3-authorize.png

ericwenger-pm committed:
NIH-consent-Gen3-images

add NIH consent and Gen3 authorization images

ericwenger-pm committed:
Update rna_seq_9.md

corrected path for NIH and gen3 screenshot images

ericwenger-pm committed:
Update rna_seq_9.md

corrected missing .png file extension for rna-seq-9-06-02-gen3-authorize.png

ericwenger-pm committed:
replacement for outdated rna-seq-9-14.png

rna-seq-9-14-AnVIL_transfer_ipynb.png

ericwenger-pm committed:
Update rna_seq_9.md

replaced old rna-seq-9-14.png with rna-seq-9-14-AnVIL_transfer_ipynb.png

ericwenger-pm committed:
Update rna_seq_9.md

remove extra ] after Portal

raynamharris committed:
Merge branch 'dev' into Cavatica-RNA-seq-training-updates

raynamharris committed:
html to markdown formatting for image test

raynamharris committed:
html to mk image links test 2

raynamharris committed:
image size test

raynamharris committed:
all html images updated to mk

raynamharris committed:
fix links, add white space

raynamharris committed:
fixes

raynamharris committed:
rm {: align=right width=95%}

raynamharris committed:
Merge branch 'dev' into Cavatica-RNA-seq-training-updates
